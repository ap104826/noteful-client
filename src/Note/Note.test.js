import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Link from './Link'

describe(`Link component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .Link by default', () => {
    const wrapper = shallow(<Link />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Link given props', () => {
    const wrapper = shallow(<Link {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
