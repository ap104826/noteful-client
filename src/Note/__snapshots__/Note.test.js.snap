// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders a .Link by default 1`] = `
<div
  className="Link"
>
  <h2
    className="Link__title"
  >
    <Link
      replace={false}
      to="/link/undefined"
    />
  </h2>
  <button
    className="Link__delete"
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
  <div
    className="Link__dates"
  >
    <div
      className="Link__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        Invalid Date
      </span>
    </div>
  </div>
</div>
`;

exports[`Link component renders the Link given props 1`] = `
<div
  className="Link"
>
  <h2
    className="Link__title"
  >
    <Link
      replace={false}
      to="/link/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Link__delete"
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
  <div
    className="Link__dates"
  >
    <div
      className="Link__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        15th Jan 2019
      </span>
    </div>
  </div>
</div>
`;
